Mu2eG4/src/Mu2eG4WorkerRunManager.cc

```diff
@@ -40,37 +40,53 @@
 #include "G4HadronicProcessStore.hh"
 
 //Other includes
-#include "CLHEP/Random/JamesRandom.h"
+#include <string>
 #include <atomic>
+#include "CLHEP/Random/JamesRandom.h"
 
+//Crypto++ includes
+#include "sha3.h"
+#include "hex.h"
+#include "files.h"
 
 using namespace std;
 
 namespace {
-  std::atomic<int> thread_counter{0};
-
+  atomic<int> thread_counter{0};
   int get_new_thread_index() { return thread_counter++; }
-
   thread_local int s_thread_index = get_new_thread_index();
-
   int getThreadIndex() { return s_thread_index; }
-
-
+  
+  
+  string CryptoPP_Hash(const string msg){
+    
+    CryptoPP::SHA3_224 hash;
+    string digest, str_hex;
+    
+    //digest is 56 hex numbers, or 28 pairs of hex numbers = 28 bytes
+    //compute the digest and keep 8 bytes of it
+    CryptoPP::StringSource(msg, true, new CryptoPP::HashFilter(hash, new CryptoPP::StringSink(digest), false, 8));
+    CryptoPP::CRYPTOPP_DLL HexEncoder encoder_out(new CryptoPP::StringSink(str_hex));
+    CryptoPP::StringSource(digest, true, new CryptoPP::Redirector(encoder_out));
```

Comment:
Same question as before:  this looks like new-ing some objects and never deleting them.  How is memory managed here?